lib/compile/transpile.js

```diff
@@ -7,26 +7,48 @@ var extd = require("../extended"),
     constraintMatcher = require("../constraintMatcher"),
     parser = require("../parser");
 
-function definedToJs(options) {
+function definedToJs(options, extend) {
     /*jshint evil:true*/
     options = isString(options) ? new Function("return " + options + ";")() : options;
-    var ret = ["(function(){"], value;
-
+    var ret = ["(function(flow){"], value;
+    
     if (options.hasOwnProperty("constructor") && "function" === typeof options.constructor) {
         ret.push("var Defined = " + options.constructor.toString() + ";");
     } else {
         ret.push("var Defined = function(opts){ for(var i in opts){if(opts.hasOwnProperty(i)){this[i] = opts[i];}}};");
     }
-    ret.push("var proto = Defined.prototype;");
+    if (extend) {
+        ret.push("var proto = Object.create(flow.getDefined('" + extend + "').prototype); proto.constructor = Defined; Defined.prototype = proto;");
+    }
+    else {
+        ret.push("var proto = Defined.prototype;");
+    }
+    
     for (var key in options) {
-        if (options.hasOwnProperty(key)) {
+        var str;
+        if (options.hasOwnProperty(key) && key !== 'constructor') {
             value = options[key];
-            ret.push("proto." + key + " = " + (extd.isFunction(value) ? value.toString() : extd.format("%j", value)) + ";");
+            str = ("proto." + key + " = ");
+            if (extd.isFunction(value)) {
+                str += value.toString();
+            }
+            else {
+                if (value instanceof Array) {
+                    str += JSON.stringify(value);
+                }
+                else {
+                    str += extd.format("%j", value);
+                }
```

Comment:
You should change this to use `JSON.stringify` or format since they are doing the same thing.